tests/react/async2.test.tsx

```diff
@@ -141,42 +141,3 @@ describe('atom read function setSelf option test', () => {
     await findByText('text: hello1')
   })
 })
-
-describe('new async atom dependency behavior in v2', () => {
-  it('should override promise with a dependency', async () => {
```

Comment:
This no longer is valid. I thought it's a good addition, but turns out it has worked because of the buggy logic.